sdk/tests/scripts/generate_fixtures.py

```diff
@@ -0,0 +1,197 @@
+# Copyright 2022 Adobe. All rights reserved.
+# This file is licensed to you under the Apache License,
+# Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+# or the MIT license (http://opensource.org/licenses/MIT),
+# at your option.
+
+# Unless required by applicable law or agreed to in writing,
+# this software is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or
+# implied. See the LICENSE-MIT and LICENSE-APACHE files for the
+# specific language governing permissions and limitations under
+# each license.
+
+import subprocess
+import os
+
+def get_fixtures_directory():
+    file_directory = os.path.dirname(os.path.realpath(__file__))
+    root_directory = subprocess.Popen(["git", "rev-parse", "--show-toplevel"], stdout=subprocess.PIPE, cwd=file_directory).communicate()[0].rstrip().decode("utf-8")
+    path_to_fixtures = os.path.join(*["sdk", "tests", "fixtures", "rustls"])
+    return os.path.join(root_directory, path_to_fixtures)
+
+FIXTURES_SPECIFICS = {
+    "rs256": {
+        "sha": "-sha256",
+        "algorithm": "RSA",
+    },
+    "rs384": {
+        "sha": "-sha384",
+        "algorithm": "RSA",
+    },
+    "rs512": {
+        "sha": "-sha512",
+        "algorithm": "RSA",
+    },
+    "ps256": {
+        "sha": "-sha256",
+        "algorithm": "RSA-PSS",
+    },
+    "ps384": {
+        "sha": "-sha384",
+        "algorithm": "RSA-PSS",
+    },
+    "ps512": {
+        "sha": "-sha512",
+        "algorithm": "RSA-PSS",
+    },
+    "es256": {
+        "sha": "-sha256",
+        "algorithm": "EC -pkeyopt ec_paramgen_curve:P-256", # TODO: leave stone age.
+    },
+    "es384": {
+        "sha": "-sha384",
+        "algorithm": "EC -pkeyopt ec_paramgen_curve:P-384",
+    },
+    # "ed25519": {
+    #     "sha": "-sha512",
+    #     "algorithm": "ED25519",
+    # },
+}
+
+def get_fixture_data(test_id):
+    fixtures_dir = get_fixtures_directory()
+
+    return {**{
+        "private_key": os.path.join(fixtures_dir, "%s_private_key.pem" % test_id),
+        "public_key": os.path.join(fixtures_dir, "%s_public_key.pem" % test_id),
+        "public_key_der": os.path.join(fixtures_dir, "%s_public_key.der" % test_id),
+        "signed_data": os.path.join(fixtures_dir, "%s_signed_data.sign" % test_id),
+        "fixtures_data": os.path.join(fixtures_dir, "../data.data"),
+    }, **FIXTURES_SPECIFICS[test_id]}
+
+def generate_fixtures_rsa(data):
+    cmds = [
+        "openssl genpkey -algorithm {algorithm} -out {private_key}".format(**data), # generate private key
```

Comment:
The actual test fixtures generation is actually a side effect, I think : this was documentation for me.

Reply:
We now provide sample certs for all signature types as to remove the need for a developer just trying the SDK from needing OpenSLL to generate certs.

Reply:
Added to the chopping block.

Reply:
New tests were a god send. Thanks!